File: test/09_test.dart
import 'dart:io';
import 'package:test/test.dart';
import 'package:advent_of_code_2020/09.dart';

void main() {
  String contents = new File('./data/09.txt').readAsStringSync();

  String example = '35\n20\n15\n25\n47\n40\n62\n55\n65\n95\n102\n117\n150\n182\n127\n219\n299\n277\n309\n576';
  
  group('A', () {
    test('Example 1', () {
      expect(A09(example, windowSize: 5), equals(127));
    });
    test('Solution', () {
      expect(A09(contents), equals(10884537));
    });
  });

  group('B', () {
    test('Example 1', () {
      expect(B09(example, windowSize: 5), equals(62));
    });
    test('Solution', () {
      expect(B09(contents), equals(1261309));
    });
  });
}

File: test/09.test.js
const fs = require('fs');

const { A09, B09 } = require('../lib/09');
const input = fs.readFileSync('./data/09.txt', 'utf-8');

const example = `35
20
15
25
47
40
62
55
65
95
102
117
150
182
127
219
299
277
309
576`;

describe('09', () => {
	describe('A', () => {
		it('Example 1', () => {
			expect(A09(example, 5)).toBe(127);
		});

		it('Solution', () => {
			expect(A09(input)).toBe(10884537);
		});
	});

	describe('B', () => {
		it('Example 1', () => {
			expect(B09(example, 5)).toBe(62);
		});

		it('Solution', () => {
			expect(B09(input)).toBe(1261309);
		});
	});
});
